File: core/CountNumbersWithUniqueDigits/CountNumbersWithUniqueDigits.java
package com.dbc;

public class CountNumbersWithUniqueDigits {
    public int countNumbersWithUniqueDigits(int n) {
        if (n == 0) return 1;
        int res = 1;
        for (int i = 1; i < n + 1; i++) {
            int start = 9, mid = 9;
            int remain = i;
            while (remain > 1) {
                mid *= start;
                remain--;
                start--;
            }
            res += mid;
        }
        return res;
    }
}

File: core/CountNumbersWithUniqueDigits/CountNumbersWithUniqueDigits.py
def countNumbersWithUniqueDigits(n: int) -> int:
    if n == 0:
        return 1
    res = 1
    for i in range(1, n + 1):
        mid, start = 9, 9
        while i > 1:
            mid *= start
            i -= 1
            start -= 1

        res += mid

    return res


if __name__ == '__main__':
    print(countNumbersWithUniqueDigits(3))
